### PR TITLE
Add error msg to tensordot operation when ndim==0

### LINKED ISSUE #332: The error message on 0D matmul is obscure and does not match NumPy.
**Describe the bug**
The error message on 0D matmul is obscure and does not match NumPy.

**To Reproduce**
```python
Python 3.7.6 (default, Jan  8 2020, 13:42:34)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sparse
>>> import numpy as np
>>> a = np.array(0)
>>> b = sparse.COO(a)
>>> b + b
0
>>> b
<COO: shape=(), dtype=int64, nnz=1, fill_value=0>
>>> b @ b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/hameerabbasi/Quansight/sparse/sparse/_coo/core.py", line 1649, in __matmul__
    return matmul(self, other)
  File "/Users/hameerabbasi/Quansight/sparse/sparse/_coo/common.py", line 203, in matmul
    return dot(a, b)
  File "/Users/hameerabbasi/Quansight/sparse/sparse/_coo/common.py", line 287, in dot
    return tensordot(a, b, axes=(a_axis, b_axis))
  File "/Users/hameerabbasi/Quansight/sparse/sparse/_coo/common.py", line 131, in tensordot
    if as_[axes_a[k]] != bs[axes_b[k]]:
IndexError: tuple index out of range
>>> a @ a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: matmul: Input operand 0 does not have enough dimensions (has 0, gufunc core with signature (n?,k),(k,m?)->(n?,m?) requires 1)
```

**Expected behavior**
The error message should match NumPy.

**System**
 - macOS 10.15.3
 - `sparse` version: `0.9.1+18.gc94927c`
 - NumPy version: 1.18.1
 - Numba version: '0.46.0'

**Additional context**

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #332

### CODE REVIEW

File: sparse/tests/test_coo.py [hunk 1]
```diff
@@ -343,6 +343,14 @@ def test_tensordot_empty():
     assert_eq(np.tensordot(x1, x2), sparse.tensordot(s1, s2))
 
 
+def test_tensordot_valueerror():
+    x1 = np.array(1)
+    x2 = np.array(1)
```
Comment (hameerabbasi) [inline sparse/tests/test_coo.py]:
```suggestion
    x1 = sparse.COO(np.array(1))
    x2 = sparse.COO(np.array(1))
```

File: sparse/_coo/common.py [hunk 2]
```diff
@@ -124,6 +124,9 @@ def tensordot(a, b, axes=2):
     bs = b.shape
     ndb = b.ndim
     equal = True
+    if nda == 0 or ndb == 0:
+        pos = 0 if nda == 0 else 1
```
Comment (hameerabbasi) [inline sparse/_coo/common.py]:
```suggestion
        pos = int(nda == 0)
```

File: sparse/tests/test_coo.py (hunk 1 above)
Comment (guilhermeleobas) [reply sparse/tests/test_coo.py]: Oops, my bad on this one.

File: sparse/_coo/common.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def tensordot(a, b, axes=2):
     ndb = b.ndim
     equal = True
     if nda == 0 or ndb == 0:
-        pos = 0 if nda == 0 else 1
+        pos = int(nda == 0)
```
Comment (hameerabbasi) [inline sparse/_coo/common.py]:
```suggestion
        pos = int(nda != 0)
```

Sorry, my bad on this one.

**Comment by hameerabbasi** [comment]
Thanks, @guilhermeleobas!

**Pull request merged**